test/unit/repeated_syncs.c

```diff
@@ -30,36 +30,50 @@
 
 #define NREPS 50
 
+#ifdef ENABLE_DEPRECATED_TESTS
 long sync_psync0[SHMEM_BARRIER_SYNC_SIZE];
 long sync_psync1[SHMEM_BARRIER_SYNC_SIZE];
+#endif
 
 int main(void)
 {
-    int i, me, npes;
+    int i, me, npes, n_loops;
 
     shmem_init();
 
     me = shmem_my_pe();
     npes = shmem_n_pes();
 
+#ifdef ENABLE_DEPRECATED_TESTS
     for (i = 0; i < SHMEM_BARRIER_SYNC_SIZE; i++) {
         sync_psync0[i] = SHMEM_SYNC_VALUE;
         sync_psync1[i] = SHMEM_SYNC_VALUE;
     }
+    n_loops = me;
+#else
+    n_loops = 1;
+#endif
 
     shmem_sync_all();
 
     /* A total of npes tests are performed, where the active set in each test
-     * includes PEs i..npes-1 */
-    for (i = 0; i <= me; i++) {
+     * includes PEs i..npes-1.
+     * Teams must share the same value for "start" (according to the OpenSHMEM
+     * specification), so in that case sync across SHMEM_TEAM_WORLD. */
+    for (i = 0; i <= n_loops; i++) {
```

Comment:
This is confusing. Do you want to iterate n_loops times or n_loops + 1 (due to the <=) ? When you set n_loops = 1 it is looping two times.

Reply:
Yes, I see the confusion.  I was thinking `n_loops + 1` is fine (2x) for `team_sync`.   So if we make this change:
```suggestion
    for (i = 0; i < n_loops; i++) {
```
but then we kinda need to set `nloops = me + 1` so PE 0 does not sync with "PE -1", which is also pretty confusing right?

So what do you suggest?  I think we can run the outer loop several more times with `team_sync`, say 10 or 11 times (prob much more) and it seems quick enough for a unit test.  Would that help?  Say, if `n_loops` is 9 or 10?

Reply:
For `team_sync`, we could also skip the outer loop entirely and just do `NREPS` if that makes sense...